### PR TITLE
fix: youtube videos not playing

### PR DESCRIPTION
**Pull request by GeronimoJohn**
# Description

Youtube videos are not loading on journeys, and cannot be played. This happens on both iPhone and iPad, for both regular video blocks and background videos. The card will either display an infinite spinning animation or just display the thumbnail image.

**The current fix is temporary** and will further look into it during this cycle's (Cycle 8) cooldown. The current fix is to disable autoplay for youtube videos and card background videos on iOS devices. On a card background, an image should be rendered instead of a video. 

[- Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/29689605/todos/5431672006)

# How should this PR be QA Tested?

1. View the [Video QA Journey](https://journeys-git-22-08-jg-fix-youtube-autoplay-jesusfilm.vercel.app/video-qa)
2. Run through each step and check off relevant tasks

### CoverBlock

- [x] autoplays on iOS
- [ ] autoplays on Android
- [x] autoplays on Web

### No CoverBlock

- [x] autoplays on iOS
- [x] autoplays on Android
- [x] autoplays on Web

### CoverBlock

- [x] show cover block image on iOS
- [x] autoplays on Android
- [x] autoplays on Web

### No CoverBlock

- [x] show youtube poster image on iOS
- [x] autoplays on Android
- [x] autoplays on Web

### CoverBlock

- [x] autoplays on iOS
- [x] autoplays on Android
- [x] autoplays on Web

### No CoverBlock

- [x] autoplays on iOS
- [x] autoplays on Android
- [x] autoplays on Web

### CoverBlock

- [x] plays on iOS
- [x] plays on Android
- [x] plays on Web

### No CoverBlock

- [x] plays on iOS
- [x] plays on Android
- [x] plays on Web

### CoverBlock

- [x] plays on iOS (no autoplay)
- [x] autoplays on Android
- [x] autoplays on Web

### No CoverBlock

- [x] plays on iOS (no autoplay)
- [x] autoplays on Android
- [x] autoplays on Web

### CoverBlock

- [x] plays on iOS
- [x] plays on Android
- [x] plays on Web

### No CoverBlock

- [x] plays on iOS
- [x] plays on Android
- [x] plays on Web

# Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged into main

### CODE REVIEW

**Comment by tataihono** [comment]
- [ ] can play an internal video on a card without an image and autoplay disabled on iOS, android, and web
- [ ] can play an internal video on a card without an image and autoplay enabled on iOS, android, and web
- [ ] can play an internal video on a card with an image and autoplay disabled on iOS, android, and web
- [ ] can play an internal video on a card with an image and autoplay enabled on iOS, android, and web
- [ ] can play a youtube video on a card without an image and autoplay disabled on iOS, android, and web
- [ ] can play a youtube video on a card without an image and autoplay enabled on iOS, android, and web
- [ ] can play a youtube video on a card with an image and autoplay disabled on iOS, android, and web
- [ ] can play a youtube video on a card with an image and autoplay enabled on iOS, android, and web

**Comment by GeronimoJohn** [comment]
There's always going to be an image with the recent changes that we've added to video.tsx. I've broken down to do to each platform to better understand what's broken and where

Web
- [ ] can play an internal video on a card with an image and autoplay disabled
- [ ] can play an internal video on a card with an image and autoplay enabled
- [ ] can play a youtube video on a background video with an image and autoplay enabled
- [ ] can play a youtube video on a background video with an image and autoplay disabled

IOS
- [ ] can play an internal video on a card with an image and autoplay disabled
- [ ] can play an internal video on a card with an image and autoplay enabled
- [ ] can play a youtube video on a background video with an image and autoplay enabled
- [ ] can play a youtube video on a background video with an image and autoplay disabled

Android
- [ ] can play an internal video on a card with an image and autoplay disabled
- [ ] can play an internal video on a card with an image and autoplay enabled
- [ ] can play a youtube video on a background video with an image and autoplay enabled
- [ ] can play a youtube video on a background video with an image and autoplay disabled

File: libs/journeys/ui/src/components/Card/ContainedCover/ContainedCover.tsx [hunk 1]
```diff
@@ -32,17 +32,21 @@ export function ContainedCover({
   const theme = useTheme()
   const [loading, setLoading] = useState(true)
 
+  const isYouTube = videoBlock?.source === VideoBlockSource.youTube
+
   useEffect(() => {
     if (videoRef.current != null) {
       playerRef.current = videojs(videoRef.current, {
```
Comment (tataihono) [inline libs/journeys/ui/src/components/Card/ContainedCover/ContainedCover.tsx:39]:
```suggestion
      // autoplay when video is YouTube on iOS does not work. We should disable autoplay in that case.
      const isYouTubeAndiOS = isYouTube && /iPhone|iPad|iPod/i.test(navigator?.userAgent)
      playerRef.current = videojs(videoRef.current, {
```

File: libs/journeys/ui/src/components/Card/ContainedCover/ContainedCover.tsx [hunk 2]
```diff
@@ -32,17 +32,21 @@ export function ContainedCover({
...
-        autoplay: true,
+        autoplay: !(
+          isYouTube && /iPhone|iPad|iPod/i.test(navigator?.userAgent)
+        ),
```
Comment (tataihono) [inline libs/journeys/ui/src/components/Card/ContainedCover/ContainedCover.tsx:42]:
```suggestion
        autoplay: !isYouTubeAndiOS,
```

File: libs/journeys/ui/src/components/Card/ContainedCover/ContainedCover.tsx [hunk 3]
```diff
@@ -32,17 +32,21 @@ export function ContainedCover({
...
         controls: false,
         preload: 'metadata',
         userActions: {
           hotkeys: false,
           doubleClick: false
         },
-        muted: true,
+        muted: !(isYouTube && /iPhone|iPad|iPod/i.test(navigator?.userAgent)),
```
Comment (tataihono) [inline libs/journeys/ui/src/components/Card/ContainedCover/ContainedCover.tsx:49]:
```suggestion
        muted: true,
```

File: libs/journeys/ui/src/components/Video/Video.tsx [hunk 4]
```diff
@@ -183,6 +180,16 @@ export function Video({
           '&:hover': {
             color: VIDEO_FOREGROUND_COLOR
           }
+        },
+        // renders big play button for youtube videos on IOS devices
```
Comment (tataihono) [inline libs/journeys/ui/src/components/Video/Video.tsx:184]:
```suggestion
        // renders big play button for youtube videos on iOS devices
```

File: libs/journeys/ui/src/components/Video/Video.tsx [hunk 5]
```diff
@@ -211,7 +218,7 @@ export function Video({
               )}
             {source === VideoBlockSource.youTube && (
               <source
-                src={`https://www.youtube.com/watch?v=${videoId}`}
+                src={`https://www.youtube.com/embed/watch?v=${videoId}`}
```
Comment (tataihono) [inline libs/journeys/ui/src/components/Video/Video.tsx:221]:
```suggestion
                src={`https://www.youtube.com/watch?v=${videoId}`}
```

**Review by tataihono — CHANGES_REQUESTED** [review]

**Review by tataihono — APPROVED** [review]

**Pull request merged**